src/extension/background-script/actions/webbtc/getAddress.ts

```diff
@@ -0,0 +1,60 @@
+import { getTaprootAddressFromPrivateKey } from "~/common/lib/btc";
+import { decryptData } from "~/common/lib/crypto";
+import {
+  BTC_TAPROOT_DERIVATION_PATH,
+  BTC_TAPROOT_DERIVATION_PATH_REGTEST,
+  derivePrivateKey,
+  derivePublicKey,
+} from "~/common/lib/mnemonic";
+import state from "~/extension/background-script/state";
+import { BitcoinAddress, MessageGetAddress } from "~/types";
+
+const getAddress = async (message: MessageGetAddress) => {
+  try {
+    // TODO: is this the correct way to decrypt the mnmenonic?
+    const password = await state.getState().password();
+    if (!password) {
+      throw new Error("No password set");
+    }
+    const account = await state.getState().getAccount();
+    if (!account) {
+      throw new Error("No account selected");
+    }
+    if (!account.mnemonic) {
+      throw new Error("No mnemonic set");
+    }
+    const mnemonic = decryptData(account.mnemonic, password);
```

Comment:
maybe we do this similar and move this to the `state` that the handling of the decryption is only done in one place? I think with the nostr key we do it similarly.

Reply:
Edit: I see your suggestion further down with `state.getState().getAccountKey();`